tests/ceph_installer/test_disconnected_install.py

```diff
@@ -0,0 +1,106 @@
+from cephci.prereq import setup_private_container_registry
+from cephci.utils.configure import exec_cephadm_preflight, setup_ssh_keys
+from cli.exceptions import NotSupportedError
+from cli.utilities.packages import Package, SubscriptionManager
+from utility.log import Log
+from utility.utils import get_cephci_config
+
+log = Log(__name__)
+
+RHCEPH_5_REPO = "rhceph-5-tools-for-rhel-8-x86_64-rpms"
+PRIVATE_REG_USER = "myregistryusername"
+PRIVATE_REG_PASS = "myregistrypassword"
+PORT = "5000"
+
+
+class ClusterCustomContainerError(Exception):
+    pass
+
+
+def get_registry_details(config):
+    """Get registry and registry credentials
+    Args:
+        config (dict): Config parameters
+    """
+    build_type = "ibm" if config.get("ibm_build") else "rh"
+    con = get_cephci_config()
+    reg_cred = con.get(f"{build_type}_registry_credentials", {})
+    return reg_cred.get("registry"), reg_cred.get("username"), reg_cred.get("password")
+
+
+def run(ceph_cluster, **kw):
+    """Verify re-deploying of monitoring stack with custom images
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get config file and node
+    config = kw.get("config")
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    mon_node = ceph_cluster.get_nodes(role="mon")[0]
+
+    rhbuild = config.get("rhbuild")
+    bootstrap_image = config.get("bootstrap_image")
+
+    # Get the rhbuild
+    if rhbuild.startswith("7"):
+        rhbuild = "reef"
+    elif rhbuild.startswith("6"):
+        rhbuild = "quincy"
+    elif rhbuild.startswith("5"):
+        rhbuild = "pacific"
+    else:
+        raise NotSupportedError(f"Unsupported version: {rhbuild}")
+
+    # Install cephadm and cephadm-ansible
+    SubscriptionManager(installer).repos.enable(RHCEPH_5_REPO)
```

Comment:
Why only rhceph5? shouldn't it depends on rhbuild ?

Reply:
Made it generic. Now accepting it from config